config/routes.rb

```diff
@@ -7,6 +7,7 @@
 
   post '/companies/update_companies_list', to: 'companies#update_companies_list'
   post '/companies/update_companies_details', to: 'companies#update_companies_details'
+  post '/companies/update_company_shareholders', to: 'companies#update_company_shareholders'
```

Comment:
why not pass the company ID this way?

```suggestion
  post '/companies/:id/update_company_shareholders', to: 'companies#update_company_shareholders'
```

Reply:
That way looks good I was thinking about in another branch.
The other question is: should I pass `:id` only here and controller should find object in database and fetch `:isin` , or it is ok to pass `:isin` as well like it is?

Reply:
I would pass company `id` here and also I would pass company `id` to `CompanyShareholdersUpdator#call` as well.

Note: In general you could remove `id` column from `companies` table and use `isin` as a primary key. But, since you have `id`, I would use it here.

Reply:
Good to know for the future, I prefer to pass `:isin` to `CompanyShareholdersUpdator#call`, in some cases `:id` is changing, it probably will not happens during shareholders updating operation,  but I will think about removing `id` column from `companies` and set `:isin` as primary key